### ISSUE TITLE
Attempt to launch stack with ContentUri fails to package local ref to content

### ISSUE THREAD

**Issue by thetumper** [Sceptre/sceptre#1290]
I'm attempting to create a lambda layer with a sceptre cloudformation template, with a local ref to the content.  This is failing when calling sceptre launch.

### Your environment
* version of sceptre (sceptre --version) --> 3.2.0
* version of python (python --version) --> 
* which OS/distro

### Steps to reproduce
The template looks like this.

```
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Description: My Application Content
Metadata:
  version: 1.0.0
Parameters:
  ApplicationName:
    Type: String
    Description: Application Name
    Default: myapp
  EnvironmentName:
    Description: Environment name
    Type: String
    Default: dev
Resources:
  ContentLayer:
    Type: AWS::Serverless::LayerVersion
    Properties:
      CompatibleRuntimes:
        - nodejs14.x
        - nodejs16.x
      ContentUri: ../../../mycontent/
      Description: Layer containing additional content
      RetentionPolicy: Retain
  LayerParameter:
    Type: AWS::SSM::Parameter
    Properties:
      Description: ARN for latest Content Layer
      Name: !Sub '/${ApplicationName}/${EnvironmentName}/content'
      Type: String
      Value: !Ref ContentLayer
```

### Expected behaviour
According to https://docs.aws.amazon.com/cli/latest/reference/cloudformation/package.html, the ContentUri property is valid for a resource of type AWS::Serverless::LayerVersion, so it would be expected to operate by executing that packaging behavior, as part of the launch process.

### Actual behaviour
The sceptre launch command fails, with the error:

Transform AWS::Serverless-2016-10-31 failed with: Invalid Serverless Application Specification document. Number of errors found: 1. Resource with id [ContentLayerd315a6d0c4] is invalid. 'ContentUri' is not a valid S3 Uri of the form 's3://bucket/key' with optional versionId query parameter.

**Comment by jfalkenstein** [comment]
Hi! What you're attempting to do isn't supported by CloudFormation. Cloudformation requires an S3 path for CodeUri. It doesn't support local filepaths. What DOES support what you're trying to do is SAM CLI. There is a Sceptre Template Handler that allows you to use SAM templates (similar to what you're trying to do). Check it out here: https://pypi.org/project/sceptre-sam-handler/

**Comment by thetumper** [comment]
I could be misunderstanding, but it sure sounds like Cloudformation supports this, via its package command, for specific properties.  It's package process takes the locally-referenced resource and uploads to S3.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-cli-package.html
https://awscli.amazonaws.com/v2/documentation/api/latest/reference/cloudformation/package.html

Not that it necessarily means that Sceptre should also support it.  But it sure would ease the way for this sort of thing, and (at least in my mind) is an example of the reason for choosing Sceptre vs. just using raw cloudformation.

FWIW, I'm specifically also trying to avoid adding SAM into my mix, because I use Serverless framework for my lambdas already, and don't want to switch over to SAM at this point...or have to maintain yet another tool in the chain, if I could use Sceptre.  :)

**Comment by zaro0508** [comment]
hmm, that error message is coming straight from AWS so I'm guessing that @jfalkenstein is correct that it's not supported by cloudformation.  You can verify that by trying to deploy that template without Sceptre, SAM or serverless framework.  

I think you are getting that error because you are using sceptre to deploy it as straight cloudformation.  If you use either SAM (without sceptre), sceptre (with sceptre-sam-handler) or the serverless framework then it will work.  Basically if you want to use sceptre you need to deploy with sceptre-sam-handler.  Another option is to create a custom sceptre serverless framework handler then use sceptre with this newly created handler.

**Comment by thetumper** [comment]
Many thanks for the responses!  They did help me understand what was going on.  I ended up using an additional job in my pipeline to zip up, then use the AWS CLI to upload to S3.  Along with that, I changed from using AWS::Serverless::Layer (SAM template), to using the AWS::Lambda::Layer cloudformation template.  The latter takes ContentUri, which references the zip I uploaded to S3, vs. the former (SAM/serverless) Content property.
